script.js

```diff
@@ -259,4 +259,37 @@ form.addEventListener('submit', (e) => {
     span.textContent = 'Email should be in lowercase';
     error.appendChild(span);
   }
-});
\ No newline at end of file
+});
+
+// localstorage
+
+const x = {
+  name: '',
+  email: '',
+  text: '',
+};
+let y = '';
+function store() {
+  y = JSON.stringify(x);
+  localStorage.setItem('form', y);
+}
+form.elements.user_name.addEventListener('change', () => {
+  x.name = form.elements.user_name.value;
+  store();
+});
+
+form.elements.user_email.addEventListener('change', () => {
+  x.email = form.elements.user_email.value;
+  store();
+});
+form.elements.user_message.addEventListener('change', () => {
+  x.text = form.elements.user_message.value;
+  store();
+});
+
+if (localStorage.getItem('form')) {
+  const get = JSON.parse(localStorage.getItem('form'));
+  form.elements.user_name.value = get.name;
+  form.elements.user_email.value = get.email;
+  form.elements.user_message.value = get.text;
+}
```

Comment:
- [ OPTIONAL] I would suggest using the **window.onload or DOMContentLoaded** event to determine when this code should be run. This event listener will only execute when the page is loaded or refreshed. What is your take 🤔 ?